interfaces/builtin/dbus.go

```diff
@@ -398,7 +425,17 @@ func (iface *DbusInterface) SanitizeSlot(slot *interfaces.Slot) error {
 	}
 
 	_, _, err := iface.getAttribs(slot.Attrs)
-	return err
+	if err != nil {
+		return err
+	}
+
+	// FIXME: also check that the dbus name is not already taken
+	//        by an existing snap
```

Comment:
The FIXME is for if a snap has taken it, not if anything has taken it. snapd has all the information for this: specifically all the snaps on the system with connected dbus interfaces and the attributes of those interfaces.

I don't think we want to query DBus only if we want to make this generalizable beyond activatable names because it would be useful if a non-activatable names were also checked. Right now, those snaps simply fail to start, but we could do something better if we were smarter with existing connections. snapd could additionally look at ListActivatableNames for the activated services to check for non-snap activated service collisions.